tokio/src/runtime/scheduler/multi_thread/worker.rs

```diff
@@ -684,7 +688,7 @@ impl Context {
     /// after all the IOs get dispatched
     fn park(&self, mut core: Box<Core>) -> Box<Core> {
         if let Some(f) = &self.worker.handle.shared.config.before_park {
```

Comment:
It seems better to me if this check were inside the `if core.transition_to_parked()` check below, to avoid generating spurious callbacks.  However it looks like it might be this way for tracing?